core/src/main/kotlin/io/saagie/updatarium/model/ChangeSet.kt

```diff
@@ -69,6 +71,21 @@ data class ChangeSet(
             maybeError?.let { error ->
                 listOf(ChangeSetError(this, error))
             } ?: emptyList()
+        } else {
+            logger.info { "$executionId already executed" }
+            emptyList()
         }
     }
+
+    /**
+     * Check if the change set must be run, either because it is mark as 'force' through its parameters or
+     * because it has not been already executed.
+     **/
+    private fun mustRun(executionId: String, configuration: UpdatariumConfiguration): Boolean =
+        if (parameters.force) {
+            logger.warn { "$executionId marked as forced. Will be executed" }
```

Comment:
I'm not found of log here.
I think that the mustRun should return a result that contains more information:

```kotlin
sealed class RunMode
object ForceRun :RunMode()
object FirstRun :RunMode()
object AlreadyRun :RunMode()
```

Reply:
is it necessary to introduce the runMode and pass it down to the runWithPersistEngine since we are able to persist the ChangeSetParameter in MongodbPersistEngine?

Reply:
Imagine `force=true` and `notAlreadyExecuted=true` you can choose what is most important.
IMO, in this situation I want to return `FirstRun`, and I want this information stored into the DB